src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -78,6 +77,21 @@ APPLICATION_INFO::Initialize(
     return hr;
 }
 
+VOID
+APPLICATION_INFO::ShutDown()
+{
+    if (m_pApplication != NULL)
+    {
+        AcquireSRWLockExclusive(&m_srwLock);
+        if (m_pApplication != NULL)
+        {
+            m_pApplication->ShutDown();
```

Comment:
I think we need deference the application here.

Reply:
Actually, we don't want to set the application to null here.

Reply:
the shutdown of n application may take long time while these could still be incoming requests. I did not see any code to handle such scenario.

Reply:
For out of process, a new application info will be created, so it is fine if shutdown takes a while. For inproc, there is nothing we can do as we need to recycle the worker process, so we will have to reject requests until a new worker process is created.